test/unit/Governor.js

```diff
@@ -0,0 +1,558 @@
+                    data: [txData]
+                }
+            )
+
+            truffleAssert.eventEmitted(tx, "OwnershipTransferred", e => e.previousOwner == accounts[0] && e.newOwner == accounts[1])
+        })
+    })
+
+    describe("stageUpdate", () => {
+        it("reverts when sender is not owner", async () => {
+            const {data} = await setUint256Tx("0", accounts[0])
+
+            await truffleAssert.reverts(
+                governor.stage(
+                    {
+                        target: [setUint256.address],
+                        value: ["0"],
+                        data: [data]
+                    },
+                    "0",
+                    {from: accounts[1]}
+                ),
+                "unauthorized",
+            )
+        })
+
+        it("reverts when the current block number added by the delay overflows", async () => {
+            const {data} = await setUint256Tx("1", accounts[0])
+
+            await fixture.rpc.mine()
+
+            await truffleAssert.reverts(
+                governor.stage(
+                    {
+                        target: [setUint256.address],
+                        value: ["0"],
+                        data: [data]
+                    },
+                    constants.MaxUint256
+                )
+            )
+        })
+
+        it("stage emits an UpdateStaged event", async () => {
+            const {data} = await setUint256Tx("1", accounts[0])
+
+            const tx = await governor.stage(
+                {
+                    target: [setUint256.address],
+                    value: ["0"],
+                    data: [data]
+                },
+                "5"
+            )
+
+            truffleAssert.eventEmitted(
+                tx,
+                "UpdateStaged",
+                e => e.update.target[0] == setUint256.address
+                    && e.update.value[0] == "0"
+                    && e.update.data[0] == data
+                    && e.delay.toString() == "5",
+                "UpdateStaged event not emitted correctly"
+            )
+        })
+    })
+
+    describe("batch stageUpdate", () => {
+        it("reverts when sender is not owner", async () => {
+            const {data: data0} = await setUint256Tx("0", accounts[0])
+            const {data: data1} = await setUint256Tx("1", accounts[0])
+            const {data: data2} = await setUint256Tx("5", accounts[0])
+
+            await truffleAssert.reverts(
+                governor.stage(
+                    {
+                        target: [setUint256.address, setUint256.address, setUint256.address],
+                        value: ["0", "0", "0"],
+                        data: [data0, data1, data2]
+                    },
+                    "0",
+                    {from: accounts[1]}
+                ),
+                "unauthorized"
+            )
+        })
+
+        it("stage emites an UpdateStaged event", async () => {
```

Comment:
Let's also check that the contract storage mapping for updates is updated correctly?